addons/mail/models/mail_render_mixin.py

```diff
@@ -262,16 +269,23 @@ def _render_template_qweb(self, template_src, model, res_ids,
         :notice: Experimental. Use at your own risks only.
         """
         results = dict.fromkeys(res_ids, u"")
+        if not template_src:
+            return results
 
         # prepare template variables
         variables = self._render_qweb_eval_context()
         if add_context:
             variables.update(**add_context)
 
+        is_restricted = not self._unrestricted_rendering and not self.env.is_admin() and not self.env.user.has_group('mail.group_mail_template_editor')
+
         for record in self.env[model].browse(res_ids):
             variables['object'] = record
             try:
-                render_result = self.env['ir.qweb']._render(html.fragment_fromstring(template_src), variables)
+                render_result = self.env['ir.qweb']._render(html.fragment_fromstring(template_src, create_parent='t'), variables, options={'raise_on_code': is_restricted})
```

Comment:
It seems like it doesn't work like it should
The following test failed, and it raise the error `"Only users [...] can modify dynamic templates."` even if the template does not contains python code
```python
    @users('user_rendering_restricted')
    def test_template_rendering_static(self):
        print("\n" * 10)
        model = 'res.partner'
        res_ids = self.env[model].search([], limit=1).ids
        partner = self.env[model].browse(res_ids)
        MailRenderMixin = self.env['mail.render.mixin']

        src = """<h1>This is a test</h1>"""

        result = self.env['mail.render.mixin'].with_user(self.user_admin)._render_template_qweb(
            src, partner._name, partner.ids,
        )[partner.id]
        print(result)

        print(src)
        # Failed here
        result = MailRenderMixin._render_template_qweb(src, model, res_ids)
        print(result)
```

Reply:
Fixed